catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/cache/impl/MetacardComparator.java

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.cache.impl;
+
+import static com.google.common.base.Strings.isNullOrEmpty;
+
+import ddf.catalog.data.Metacard;
+
+public class MetacardComparator {
+
+    private MetacardComparator() {
+    }
+
+    private static boolean validChecksum(Metacard cachedMetacard, Metacard updatedMetacard) {
+
+        return (!isNullOrEmpty(cachedMetacard.getAttribute(Metacard.CHECKSUM)
+                .toString()) && !isNullOrEmpty(updatedMetacard.getAttribute(Metacard.CHECKSUM)
+                .toString()));
+    }
+
+    public static boolean isSame(Metacard cachedMetacard, Metacard updatedMetacard) {
+
+        boolean result = false;
+
+        if (validChecksum(cachedMetacard, updatedMetacard)) {
+
+            result = cachedMetacard.getAttribute(Metacard.CHECKSUM)
+                    .toString()
+                    .equals(updatedMetacard.getAttribute(Metacard.CHECKSUM)
+                            .toString()) &&
+
+                    cachedMetacard.getModifiedDate()
+                            .equals(updatedMetacard.getModifiedDate()) &&
+                    cachedMetacard.getResourceURI()
+                            .equals(updatedMetacard.getResourceURI()) &&
+
+                    cachedMetacard.getContentTypeName()
+                            .equals(updatedMetacard.getContentTypeName()) &&
+
+                    cachedMetacard.getResourceSize()
+                            .equals(updatedMetacard.getResourceSize()) &&
+
+                    cachedMetacard.getCreatedDate()
+                            .equals(updatedMetacard.getCreatedDate());
+
+            result = result &&
```

Comment:
It's not used a lot, but there is the unary operator `&=` that you could use here.